std/os.zig

```diff
@@ -756,6 +756,44 @@ pub fn getEnvMap(allocator: *Allocator) !BufMap {
 
             try result.setMove(key, value);
         }
+    } else if (builtin.os == Os.wasi) {
+        var environ_count: usize = undefined;
+        var environ_buf_size: usize = undefined;
+
+        const environ_sizes_get_ret = std.os.wasi.environ_sizes_get(&environ_count, &environ_buf_size);
+        if (environ_sizes_get_ret != os.wasi.ESUCCESS) {
+            return unexpectedErrorPosix(environ_sizes_get_ret);
+        }
+
+        // TODO: Verify that the documentation is incorrect
+        // https://github.com/WebAssembly/WASI/issues/27
+        var environ = try allocator.alloc(?[*]u8, environ_count + 1);
+        defer allocator.free(environ);
+        var environ_buf = try std.heap.wasm_allocator.alloc(u8, environ_buf_size);
+        defer allocator.free(environ_buf);
+
+        const environ_get_ret = std.os.wasi.environ_get(environ.ptr, environ_buf.ptr);
+        if (environ_get_ret != os.wasi.ESUCCESS) {
+            return unexpectedErrorPosix(environ_get_ret);
+        }
+
+        for (environ) |env| {
+            if (env) |ptr| {
+                var line_i: usize = 0;
+                while (ptr[line_i] != 0 and ptr[line_i] != '=') : (line_i += 1) {}
+                if (ptr[line_i] != '=') {
```

Comment:
I think this can be an assertion. If the WASI functions don't do what they're documented to do, that's definitely going to cause undefined behavior. And `environ_get` is documented to return the environment string with the format `key=value`, right?

Reply:
The documentation doesn't say anything. wasmtime returns it in `key=value`, wasmer returns argv[0] which is definitely a bug.

Reply:
We have 2 choices here:
1. make the zig code assume the WASI host is correct and do a little bit of prediction about the specifications of the behavior (`key=value`) and allow host bugs to cause bugs in Zig code that uses them.
2. recognize and work around WASI host bugs & undocumented behavior.

Given that WASI is so brand new, my preference is (1). That means that we recommend Debug or ReleaseSafe builds for now - due to upstream/downstream bugs - and then work with the third party projects to get their bugs fixed / documentation clarified.